packages/rspeedy/plugin-react-alias/src/index.ts

```diff
@@ -180,21 +180,37 @@ export function pluginReactAlias(options: Options): RsbuildPlugin {
   }
 }
 
-export function createLazyResolver(context: string, conditionNames: string[]) {
-  let lazyExports: Record<string, string | false>
-  let resolverLazy: ResolveFunction
+export function createLazyResolver(
+  directory: string,
+  conditionNames: string[],
+) {
+  let lazyExports: Record<string, string>
+  let resolverLazy: (directory: string, request: string) => ResolveResult
 
   return async (
     request: string,
   ): Promise<string> => {
-    const { default: resolver } = await import('enhanced-resolve')
+    const { ResolverFactory } = await import('unrs-resolver')
+    const resolver = new ResolverFactory({ conditionNames })
```

Comment:
It looks like we need to create a new `ResolverFactory` for each `resolve` request. Can we move this into the `if` below?